plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -1,54 +1,72 @@
 /**
  * External dependencies
  */
+import {
+	Combobox as AriakitCombobox,
+	ComboboxLabel,
+	ComboboxPopover,
+	ComboboxProvider,
+	ComboboxItem,
+	useComboboxStore,
+} from '@ariakit/react';
+import type { ComboboxProps as AriakitComboboxProps } from '@ariakit/react';
 import classnames from 'classnames';
 import { __ } from '@wordpress/i18n';
-import { useEffect, useId, useRef } from '@wordpress/element';
-import { ComboboxControl } from 'wordpress-components';
+import {
+	useCallback,
+	useEffect,
+	useId,
+	useMemo,
+	useRef,
+	useState,
+	useTransition,
+} from '@wordpress/element';
 import { ValidationInputError } from '@woocommerce/blocks-components';
-import { isObject } from '@woocommerce/types';
 import { useDispatch, useSelect } from '@wordpress/data';
 import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
+import {
+	findBestMatchByLabel,
+	findExactMatchByLabel,
+	findMatchingSuggestions,
+} from './util';
 
 export interface ComboboxControlOption {
 	label: string;
 	value: string;
 }
 
-export interface ComboboxProps {
-	autoComplete?: string;
-	className?: string;
+type WCComboboxProps = Omit< AriakitComboboxProps, 'onChange' > & {
```

Comment:
> Out of curiosity, is there a particular reason why we're using WCComboboxProps instead of ComboboxProps here? I'm asking as I do not recall seeing other TS interfaces or types being prefixed with WC.

No reason I can argue for :) happy to change this and export. Thanks for spotting the TS error, its tough to spot them without proper ts compiling right now.

Reply:
fwiw that type consuming the props is incorrect and redundant. I won't remove it here though. i've renamed and exported the type.